src/ape/managers/chain.py

```diff
@@ -408,18 +414,82 @@ def outgoing(self) -> Iterator[ReceiptAPI]:
         """
         All outgoing transactions, from earliest to latest.
         """
-        explorer = self.provider.network.explorer
-        explorer_receipts = (
-            [r for r in explorer.get_account_transactions(self.address)] if explorer else []
-        )
-        for receipt in explorer_receipts:
-            if receipt.sender != self.address:
-                # Likely ``self.address`` is a contract.
-                # Cache the receipts by their sender instead and skip them here.
-                self.chain_manager.history.append(receipt)
-                continue
+
+        start_nonce = 0
+        stop_nonce = self.__len__() - 1  # just to cache this value
+
+        # TODO: Add ephemeral network sessional history to `ape-cache` instead,
+        #       and remove this (replace with `yield from iter(self[:len(self)])`)
+        for receipt in self.sessional:
+            if receipt.nonce < start_nonce:
+                raise QueryEngineError("Sessional history corrupted")
+
+            if receipt.nonce > start_nonce:
+                # NOTE: There's a gap in our sessional history, so fetch from query engine
+                yield from iter(self[start_nonce : receipt.nonce + 1])  # noqa: E203
 
             yield receipt
+            start_nonce = receipt.nonce + 1  # start next loop on the next item
+
+        if start_nonce != stop_nonce:
+            # NOTE: there is no more sessional history, so just return query engine iterator
+            yield from iter(self[start_nonce : stop_nonce + 1])  # noqa: E203
+
+    def query(
+        self,
+        *columns: List[str],
+        start_nonce: int = 0,
+        stop_nonce: Optional[int] = None,
```

Comment:
I wish the resulting dataframe was indexed by nonce always.
When I do:

```
account.history.query("transaction", start_once=3, stop_nonce=6)
```

it shows the result like numeric indices 0 - 3, which threw me off at first.

Reply:
Yeah I agree with this. Probably a breaking change though, issue to track for 0.7.0?

Reply:
i mean, you can't do anything with the data now.. it's just broken

Reply:
Actually wait, we added this method here, so I think we could actually do it for account history (but not for block history or event history)

Reply:
nvm, it requires siginificant restructuring to how we handle the returns from the query manager in order to create an index column, so not handling this here